Modified katello-service to support enable/disable for systemd

fftux opened the pull request:
Changes in PR #306 lay ground work for enable/disable support.  However, when enable/disable is invoked on a systemd system, it parses the command as "service-wait #{service} #{action}".  Thus, I modified the manage_services method to utilize systemctl and chkconfig where necessary.  It's crude, but it works.  Not a dev.

fftux committed:
updated manage_services

Added logic for systemctl and chkconfig.  This is necessary to support enable/disable functions for systemd and upstart.

fftux committed:
Merge pull request #1 from chopskxw/katello-service-fixes

updated manage_services

ehelms commented on katello/katello-service:
```diff
@@ -78,7 +78,19 @@ def manage_services(action)
+          puts `chkconfig #{service} on`
+        else
+          puts "#{COMMAND} #{service} #{action}"
+          puts `#{COMMAND} #{service} #{action}`
```
Is the double puts intended for each action?

fftux replied:
Without the double puts, there is not output of what is going on.  Just a blank line for each command, followed by the "Success" or failures.

ehelms commented:
Thanks for the PR @chopskxw ! This looks like it would change not only disable/enable but other commands as well? I think we would need to test whether moving away from service-wait for some services would work for us. The original intent of service-wait was due to system services deeming a service "up" or "running" even though said service was not reachable yet (e.g. waiting on a port to become available to talk to or a the webserver to be up). This was written many years ago, and systemd may properly handle this now.

fftux commented:
I can back out the use of systemctl here, and rely on chkconfig to do the work.  However, this will likely become antiquated soon rather than later.  Obviously there are a few different ways to achieve the end goal, this just seemed like the path of least resistance.  Will be happy to make some mods, if necessary.

ehelms commented:
@chopskxw What if we started with just checking for enable/disable and use systemd ? And continue using service-wait for other service actions? 

The next step would be to look at what service-wait (https://github.com/Katello/katello-packaging/blob/master/katello/service-wait) is doing and decide whether it has any antiquated ways of checking if a service is started or not given systemd on EL7 to begin moving away from it.

fftux committed:
Modified for minimal use of systemctl

Systemctl only called for enable/disable.  Service-wait still used for other functions.

fftux committed:
Corrected syntax

fixed missing end

fftux committed:
corrected syntax

bad syntax

fftux commented:
@ehelms reorganized method to check for enable/disable action first, and only run systemctl when for these functions.

ehelms commented on katello/katello-service:
```diff
@@ -78,8 +78,20 @@ def manage_services(action)
+        puts "#{COMMAND} #{service} #{action}"
+        puts `#{COMMAND} #{service} #{action}`
+      end
+    failures << service unless $?.success?
```
Looks like this should be indented to appear inside the if-else

ehelms commented:
Seems reasonable -- I will test these changes locally.

fftux committed:
fix spacing

fftux committed:
Change status behavior

Rather than modifying the list option, which should simply list the services, added method for providing brief status of services.  Full systemd style status can be obtained with the -v or --verbose option.  This can also be used in conjunction with other options.

fftux committed:
Merge pull request #2 from chopskxw/issue-238

Change status behavior

fftux committed:
Revert "Change status behavior"

fftux committed:
Merge pull request #3 from chopskxw/revert-2-issue-238

Revert "Change status behavior"

ehelms commented:
Looks and works good, Thanks @chopskxw !
